### ISSUE TITLE
Better error messages when an `@Output` property is not initialized.

### ISSUE THREAD

**Issue by pkozlowski-opensource** [angular/angular#3664]
Related: #3754 (should be fixed together)

Listing an event in a directive without creating an associated `EventEmitter` instance results in cryptic error:

> TypeError: Cannot read property 'observer' of undefined
>     at Function.ObservableWrapper.subscribe (angular2.js:19403)
>     at EventEmitterAccessor.subscribe (angular2.js:26062)
>     at AppViewManagerUtils._setUpEventEmitters (angular2.js:20932)
>     at AppViewManagerUtils._hydrateView (angular2.js:20896)
>     at AppViewManagerUtils.hydrateRootHostView (angular2.js:20814)
>     at AppViewManager.createRootHostView (angular2.js:23990)
>     at angular2.js:13594
>     at Zone.run (zone.js:113)
>     at Zone.run (angular2.js:12928)
>     at zoneBoundFn (zone.js:86)

To reproduce:

``` java
@Directive({
    selector: '[foo]'
})
class Foo {
  @Output() foo; // not initialized to = new EventEmitter();
}
```

**Comment by tomascharad** [comment]
Should this be closed as #3754 is closed?

**Comment by mhevery** [comment]
Is this still an issue? Can you fix it?

**Comment by pkozlowski-opensource** [comment]
Still an issue. Will take a stab at improving an error message.

**Comment by vicb** [comment]
@pkozlowski-opensource Have you had a chance to look at this ?

**Comment by DzmitryShylovich** [comment]
@vicb `Directive` doesn't have `events` property. close?

**Comment by vicb** [comment]
`outputs` used to be named `events`, I'll update the description

**Comment by ashish173** [comment]
Yes I also got this cryptic error message. I am going to try to fix this one.

**Comment by cedvdb** [comment]
When subscribing to the non initialized EventEmitter in the template eg:

```
    <editor (submitted)="onReply()"></editor>
   // with 
      @Output() submitted;
```

The error message is also not very precise as of where the error is coming from.

```
core.es5.js:1084 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'subscribe' of undefined
TypeError: Cannot read property 'subscribe' of undefined
```

**Comment by aldo-roman** [comment]
Hi @mhevery 
I submitted a PR for this issue some time ago. I just fixed a conflict, lint errors and tests. See #19116

Please let me know if there is something I can do. Cheers!

**Comment by splincode** [comment]
@pkozlowski-opensource Is the issue relevant?
Will this pull request be accepted?
https://github.com/angular/angular/pull/19116

**Comment by luisfelix199276** [comment]
For me the erro stopped when I changed the import statement from import { EventEmitter } from 'events'; to this import { EventEmitter } from '@angular/core'. The first option was suggested by angular as default but I think that the second one is the right one. Every thing works fine now after I changed.

@Output() variableToEmit: EventEmitter<string> = new EventEmitter();

 showsString(text: any) {
    this.variableToEmit.emit(text);
  }
